crates/wdk-build/src/utils.rs

```diff
@@ -449,6 +452,87 @@ pub(crate) fn find_max_version_in_directory<P: AsRef<Path>>(
         })
 }
 
+/// Safely sets an environment variable, panicking if not running on a Windows
+/// host.
+///
+/// This function provides a safe wrapper around [`std::env::set_var`] that
+/// became unsafe in Rust 2024 edition. It validates that the current **host**
+/// OS (where the build is running) is Windows before calling the unsafe
+/// function.
+///
+/// Note: This checks the host OS, not the target OS. If you're cross-compiling
+/// from macOS to Windows, this will still panic because the host (macOS) is not
+/// Windows.
+///
+/// # Panics
+///
+/// Panics if the current host OS is not Windows, since environment variable
+/// operations are only guaranteed to be safe on Windows according to Rust
+/// documentation.
+///
+/// # Examples
+///
+/// ```
+/// wdk_build::utils::set_var("MY_BUILD_VAR", "my_value");
+/// ```
+pub fn set_var<K, V>(key: K, value: V)
```

Comment:
could this be `pub(crate)`? we are trying to return utils module back to private before making another crates release: https://github.com/microsoft/windows-drivers-rs/issues/367